honeybadgermpc/mpc.py

```diff
@@ -64,11 +64,27 @@ def __init__(self, sid, n, t, myid, pid, send, recv, prog, config, **prog_args):
         self.GFElementFuture = type(
             'GFElementFuture', (GFElementFuture,), {'context': self})
 
+    def _get_share_id(self):
```

Comment:
Ummmm, there are some serious issues in the way we assign the `ShareId`. @rahulgovind and I had discussed an example in which assigning the order this way won't be consistent across all parties since the tasks are not executed in a deterministic order.

Why did you see a need to change this?

Reply:
Currently, shares are already assigned ids in this order-- we get the length of `self._openings` and assign `share_id` that way, then assign an opening to that slot of `self._openings` before ever yielding execution to another task. This simply adds an extra layer of overhead to understanding the code given that we use `self._openings` for no other reason today.

Conversely,  the new code makes the assignment of ids much more explicit, and also saves the memory that `self._openings` took up. We never deleted entries once they finished opening, and never actually use `self._openings` other than to assign these IDs.

 I think in the current state of `Mpc`, this change is better.

Reply:
Sounds good!